Vagrantfile

```diff
@@ -6,6 +6,196 @@ $NODEMEM=256
+
+readonly workshop_git_repo="https://github.com/ansible/lightbulb.git"
+
+#
+# Functions
+#
+
+main() {
+  info "Bootstrapping Ansible on host ${HOSTNAME}."
+
+  ensure_ansible_installed
+  install_vagrant_private_key
+  configure_ansible
+  install_inventory
+  install_lightbulb
+
+}
+
+# Ansible installation
+ensure_ansible_installed() {
+  if ! is_ansible_installed; then
+    distro=$(get_linux_distribution)
+    "install_ansible_${distro}" || die "Distribution ${distro} is not supported"
+  fi
+
+  info "Ansible version"
+  ansible --version
+}
+
+is_ansible_installed() {
+  which ansible-playbook > /dev/null 2>&1
+}
+
+# Print the Linux distribution
+get_linux_distribution() {
+
+  if [ -f '/etc/redhat-release' ]; then
+
+    # RedHat-based distributions
+    cut --fields=1 --delimiter=' ' '/etc/redhat-release' \
+      | tr "[:upper:]" "[:lower:]"
+
+  elif [ -f '/etc/lsb-release' ]; then
+
+    # Debian-based distributions
+    grep DISTRIB_ID '/etc/lsb-release' \
+      | cut --delimiter='=' --fields=2 \
+      | tr "[:upper:]" "[:lower:]"
+
+  fi
+}
+
+# Install Ansible on a Fedora system.
+# The version in the repos is fairly current, so we'll install that
+install_ansible_fedora() {
+  info "Fedora: installing Ansible from distribution repositories"
+  dnf -y install git
+}
+
+# Install Ansible on a CentOS system from EPEL
+install_ansible_centos() {
+  info "CentOS: installing Ansible from the EPEL repository"
+  yum -y install epel-release
+  yum -y install git
+}
+
+# Install Ansible on a recent Ubuntu distribution, from the PPA
+install_ansible_ubuntu() {
+  info "Ubuntu: installing Ansible from PPA"
+  # Remark: on older Ubuntu versions, it's python-software-properties
+  apt-get -y install software-properties-common
+  apt-add-repository -y ppa:ansible/ansible
+  apt-get -y update
+  apt-get -y install git
+}
+
+# Install the Vagrant private key to SSH to the other nodes
+install_vagrant_private_key() {
+  curl --output "${workshop_key}" \
+    https://raw.githubusercontent.com/hashicorp/vagrant/master/keys/vagrant
+  chown "${workshop_user}:${workshop_user}" "${workshop_key}"
+  chmod 0600 "${workshop_key}"
+}
+
+# Install an inventory file
+install_inventory() {
+  info "Installing inventory file ${workshop_inventory}"
+
+  cat > "${workshop_inventory}" << _EOF_
+[control]
+ansible ansible_host=10.42.0.2 ansible_connection=local
+
+[web]
+node-1 ansible_host=10.42.0.6
+node-2 ansible_host=10.42.0.7
+node-3 ansible_host=10.42.0.8
+
+[all:vars]
+ansible_user=vagrant
+_EOF_
+  chown "${workshop_user}:${workshop_user}" "${workshop_inventory}"
+}
+
+# This will generate an ansible.cfg file from a here document
+configure_ansible() {
+  info "Installing Ansible configuration file ${workshop_ansible_config}"
+
+  cat > "${workshop_ansible_config}" << _EOF_
+# config file for ansible -- http://ansible.com/
+# ==============================================
+
+[defaults]
+
+inventory = ./hosts.ini
+forks = 50
```

Comment:
This is too high for this lab environment (I know that in the source material from another use case) -- it should be 10 or removed entirely.